### DISCUSSION TITLE
Support for fst format

### DISCUSSION THREAD

**Discussion by hope-data-science** [duckdb/duckdb#6518] [Ideas]
I have lots of files in fst format (<https://github.com/fstpackage/fst>), because this format have high performance in read and write. Is it possible to let duckdb process this kind of file (like what it did in parquet)? 

Thanks.

**Comment by rsund** [comment]
Fst seems to be a storage format for columnar tabular data using [fstlib C++ library](https://github.com/fstpackage/fstlib). Probably at least reading could be rather straightforward thing to do, but I may be totally wrong :). On the other hand, would it be enough if you use R client to read fst there and register the obtained data.frame for DuckDB using `duckdb::duckdb_register()`?

**Comment by hope-data-science** [comment]
Reading this format in R is fast, however, I am looking for an implementation to process data on disk but not in memory. Fst is small in size and fast to read, but if I want to use the data I always have to import the data to the memory, and sometimes I have a very large dataset.

**Comment by AndyTwo2** [comment]
This would be a great benefit if we could get the support for processing fst files into duckdb, often reading those files into R to generate appropriate queries in R is the slowest part of my data analysis pipeline, and if duckdb could provide similar functionality to that outlined here: https://duckdb.org/2021/06/25/querying-parquet.html but for fst files I would be eternally grateful!

I'm no expert, but it doesn't seem conceptually all that different to the existing support for parquet files (although I'm no expert), so hopefully it's not too difficult? I'd be happy to try and help out if someone could provide details of what would need to be done to get started. A good summary of the differences between the two formats is given by the package author here: https://github.com/fstpackage/fst/issues/129

**Reply by rsund** [reply]
I would convert files to .parquet using DuckDB (read with fst, duckdb_register and then  COPY TO parquet in DuckDB).
